### PR TITLE
`functional.max_unpool`: OpInfo tests + simpler backward + forward ad + fwad over backward ad

### LINKED ISSUE #67657: Gradient tests failing for `max_unpool1d`
## 🐛 Bug

Context: While trying to add OpInfo for `max_unpool1d`, `test_fn_grad` and `test_fn_gradgrad` are failing. See the error below:

Caution: long snippet alert!!

<details>

```python
======================================================================
ERROR: test_fn_grad_nn_functional_max_unpool1d_cpu_float64 (__main__.TestGradientsCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 376, in instantiated_test
    raise rte
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 371, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 737, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/pytorch/test/test_ops.py", line 603, in test_fn_grad
    self._grad_test_helper(device, dtype, op, op.get_op())
  File "/home/krshrimali/pytorch/test/test_ops.py", line 586, in _grad_test_helper
    return self._check_helper(device, dtype, op, variant, 'gradcheck', check_forward_ad=check_forward_ad,
  File "/home/krshrimali/pytorch/test/test_ops.py", line 559, in _check_helper
    self.assertTrue(gradcheck(fn, gradcheck_args,
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 2718, in gradcheck
    return torch.autograd.gradcheck(fn, inputs, **kwargs)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1273, in gradcheck
    return _gradcheck_helper(**args)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1286, in _gradcheck_helper
    _gradcheck_real_imag(gradcheck_fn, func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 947, in _gradcheck_real_imag
    gradcheck_fn(func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1168, in _fast_gradcheck
    _check_analytical_numerical_equal(analytical_vJu, numerical_vJu, complex_indices,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1148, in _check_analytical_numerical_equal
    raise GradcheckError(_get_notallclose_msg(a, n, j, i, complex_indices, test_imag, is_forward_ad) + jacobians_str)
torch.autograd.gradcheck.GradcheckError: Jacobian mismatch for output 0 with respect to input 0,
numerical:tensor(0.4891)
analytical:tensor(0.7213)

The above quantities relating the numerical and analytical jacobians are computed
in fast mode. See: https://github.com/pytorch/pytorch/issues/53876 for more background
about fast mode. Below, we recompute numerical and analytical jacobians in slow mode:

Numerical:
 tensor([[0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 1.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 1.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.5000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         1.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 1.0000, 0.0000, 0.0000]])
Analytical:
tensor([[0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0.]])

The max per-element difference (slow mode) is: 1.0.


======================================================================
ERROR: test_fn_gradgrad_nn_functional_max_unpool1d_cpu_float64 (__main__.TestGradientsCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 376, in instantiated_test
    raise rte
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 371, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 737, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/pytorch/test/test_ops.py", line 625, in test_fn_gradgrad
    self._gradgrad_test_helper(device, dtype, op, op.get_op())
  File "/home/krshrimali/pytorch/test/test_ops.py", line 591, in _gradgrad_test_helper
    return self._check_helper(device, dtype, op, variant, 'gradgradcheck')
  File "/home/krshrimali/pytorch/test/test_ops.py", line 569, in _check_helper
    self.assertTrue(gradgradcheck(fn, gradcheck_args,
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 2738, in gradgradcheck
    return torch.autograd.gradgradcheck(fn, inputs, grad_outputs, **kwargs)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1402, in gradgradcheck
    return gradcheck(
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1273, in gradcheck
    return _gradcheck_helper(**args)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1286, in _gradcheck_helper
    _gradcheck_real_imag(gradcheck_fn, func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 947, in _gradcheck_real_imag
    gradcheck_fn(func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1166, in _fast_gradcheck
    analytical_vJu = _get_analytical_vJu_backward_mode(inputs, outputs, nondet_tol, check_grad_dtypes, all_v, all_u_dense)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 555, in _get_analytical_vJu_backward_mode
    all_vJ = _check_analytical_jacobian_attributes(inputs, output, nondet_tol, check_grad_dtypes,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 544, in _check_analytical_jacobian_attributes
    raise GradcheckError('Backward is not reentrant, i.e., running backward with '
torch.autograd.gradcheck.GradcheckError: Backward is not reentrant, i.e., running backward with same input and grad_output multiple times gives different values, although analytical gradient matches numerical gradient.The tolerance for nondeterminism was 0.0.

NOTE: If your op relies on non-deterministic operations i.e., it is listed here:
https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html
this failure might be expected.

If you are adding a new operator, please file an issue and then use one of the
workarounds. The workaround depends on how your test invokes gradcheck/gradgradcheck.
If the test
- manually invokes gradcheck/gradgradcheck, then call gradcheck/gradgradcheck
  with `nondet_tol=<tol>` as a keyword argument.
- is OpInfo-based (e.g., in test_ops.py), then modify the OpInfo for the test
  to have `gradcheck_nondet_tol=<tol>`.
- is a Module test (e.g., in common_nn.py), then modify the corresponding
  module_test entry to have `gradcheck_nondet_tol=<tol>`


======================================================================
ERROR: test_fn_grad_nn_functional_max_unpool1d_cuda_float64 (__main__.TestGradientsCUDA)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 1422, in wrapper
    method(*args, **kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 1422, in wrapper
    method(*args, **kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 376, in instantiated_test
    raise rte
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 371, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 737, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/pytorch/test/test_ops.py", line 603, in test_fn_grad
    self._grad_test_helper(device, dtype, op, op.get_op())
  File "/home/krshrimali/pytorch/test/test_ops.py", line 586, in _grad_test_helper
    return self._check_helper(device, dtype, op, variant, 'gradcheck', check_forward_ad=check_forward_ad,
  File "/home/krshrimali/pytorch/test/test_ops.py", line 559, in _check_helper
    self.assertTrue(gradcheck(fn, gradcheck_args,
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 2718, in gradcheck
    return torch.autograd.gradcheck(fn, inputs, **kwargs)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1273, in gradcheck
    return _gradcheck_helper(**args)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1286, in _gradcheck_helper
    _gradcheck_real_imag(gradcheck_fn, func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 947, in _gradcheck_real_imag
    gradcheck_fn(func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1168, in _fast_gradcheck
    _check_analytical_numerical_equal(analytical_vJu, numerical_vJu, complex_indices,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1148, in _check_analytical_numerical_equal
    raise GradcheckError(_get_notallclose_msg(a, n, j, i, complex_indices, test_imag, is_forward_ad) + jacobians_str)
torch.autograd.gradcheck.GradcheckError: Jacobian mismatch for output 0 with respect to input 0,
numerical:tensor(0.1653, device='cuda:0')
analytical:tensor(0.1880, device='cuda:0')

The above quantities relating the numerical and analytical jacobians are computed
in fast mode. See: https://github.com/pytorch/pytorch/issues/53876 for more background
about fast mode. Below, we recompute numerical and analytical jacobians in slow mode:

Numerical:
 tensor([[1.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 1.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 1.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000]], device='cuda:0')
Analytical:
tensor([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0.]], device='cuda:0')

The max per-element difference (slow mode) is: 1.0.


======================================================================
ERROR: test_fn_gradgrad_nn_functional_max_unpool1d_cuda_float64 (__main__.TestGradientsCUDA)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 1422, in wrapper
    method(*args, **kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 1422, in wrapper
    method(*args, **kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 376, in instantiated_test
    raise rte
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 371, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 737, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/pytorch/test/test_ops.py", line 625, in test_fn_gradgrad
    self._gradgrad_test_helper(device, dtype, op, op.get_op())
  File "/home/krshrimali/pytorch/test/test_ops.py", line 591, in _gradgrad_test_helper
    return self._check_helper(device, dtype, op, variant, 'gradgradcheck')
  File "/home/krshrimali/pytorch/test/test_ops.py", line 569, in _check_helper
    self.assertTrue(gradgradcheck(fn, gradcheck_args,
  File "/home/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 2738, in gradgradcheck
    return torch.autograd.gradgradcheck(fn, inputs, grad_outputs, **kwargs)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1402, in gradgradcheck
    return gradcheck(
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1273, in gradcheck
    return _gradcheck_helper(**args)
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1286, in _gradcheck_helper
    _gradcheck_real_imag(gradcheck_fn, func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 947, in _gradcheck_real_imag
    gradcheck_fn(func, func_out, tupled_inputs, outputs, eps,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1168, in _fast_gradcheck
    _check_analytical_numerical_equal(analytical_vJu, numerical_vJu, complex_indices,
  File "/home/krshrimali/pytorch/torch/autograd/gradcheck.py", line 1148, in _check_analytical_numerical_equal
    raise GradcheckError(_get_notallclose_msg(a, n, j, i, complex_indices, test_imag, is_forward_ad) + jacobians_str)
torch.autograd.gradcheck.GradcheckError: Jacobian mismatch for output 0 with respect to input 1,
numerical:tensor(0.0982, device='cuda:0')
analytical:tensor(0.0662, device='cuda:0')

The above quantities relating the numerical and analytical jacobians are computed
in fast mode. See: https://github.com/pytorch/pytorch/issues/53876 for more background
about fast mode. Below, we recompute numerical and analytical jacobians in slow mode:

Numerical:
 tensor([[1.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 1.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 1.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 1.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 1.0000, 1.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000]], device='cuda:0')
Analytical:
tensor([[1., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 1., 0., 0., 0., 0.],
        [0., 0., 1., 0., 0., 0.],
        [0., 0., 0., 1., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 1., 0.],
        [0., 0., 0., 0., 0., 0.]], device='cuda:0')

The max per-element difference (slow mode) is: 0.9999999999998899.
```

</details>

Interestingly, `test_fn_gradgrad_nn_functional_max_unpool1d_cpu_float64` fails with an error similar for non-deterministic ops, but the op is not listed here: https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html.

## To Reproduce

1. Checkout to the branch: https://github.com/krshrimali/pytorch/tree/origin/max_unpoolNd
2. Run the OpInfo tests after removing the skips for `max_unpool1d`. `python3 test/test_ops.py -k max_unpool1d`

## Expected behavior

The tests should pass.

## Additional Context

Please see https://github.com/pytorch/pytorch/pull/67328 and https://github.com/pytorch/pytorch/pull/67328#discussion_r738082715 for the discussion.

cc: @albanD @nikitaved 

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @mruberry @walterddr

### PR DESCRIPTION
**Pull request by nikitaved**
Resolves https://github.com/pytorch/pytorch/issues/67657, https://github.com/pytorch/pytorch/issues/67658, https://github.com/pytorch/pytorch/issues/67660.

These are not necessarily bugs because we cannot produce arbitrary samples coming from `max_pool` to the gradcheck's eternal satisfaction.

This PR also replaces low-level complicated backward kernels with much simpler high-level and well-tested counterparts. The replacement is also faster (before: parallel for loop, after: memory layout optimized TensorIterator's parallelization coming from `gather`).

cc @albanD @mruberry @jbschlosser @walterddr

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Formula update lgtm

**Comment by albanD** [comment]
cc @mruberry this should be ready to review the opinfo part and land.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -3289,6 +3289,85 @@ def generator():
...
+                    'padding': sample.kwargs['padding'],
+                    # output_size could be None but we specify it explicitly
+                    # to compensate for the information lose in pool due
+                    # to the floor/ceil operation used to compute the shapes
+                    'output_size': sample.input.size()
+                }
+                yield SampleInput(arg, args=(indices,), kwargs=sample_kwargs)
+
+    return list(generator())
+
```
Comment (nikitaved) [inline torch/testing/_internal/common_methods_invocations.py:3345]: No need in doing so anymore.

**Comment by nikitaved** [comment]
Ah, that would be great to finally merge.

**Comment by Lezcano** [comment]
As the OpInfo was already merged in https://github.com/pytorch/pytorch/pull/75782, we should be able to land the rest after a rebase and an OK from @albanD .

**Review by albanD — APPROVED** [review]
Thanks!
You can land when the CI is green!

**Comment by nikitaved** [comment]
Thank you, @albanD , but I will remove `max_unpool2d/3d_backward` from the codebase. Are you fine with that?

File: aten/src/ATen/native/MaxUnpooling.cpp [hunk 2]
```diff
@@ -185,132 +185,8 @@ Tensor max_unpooling3d_forward_cpu(
...
-  /* get contiguous gradOutput */
-  auto grad_output = grad_output_.contiguous();
-  auto indices = indices_.contiguous();
-
-  /* resize */
-  grad_input.resize_as_(self);
-  grad_input.zero_();
-
-  if (oW != grad_output.size(dimw) || oH != grad_output.size(dimh) || oT != grad_output.size(dimt)) {
-    AT_ERROR(
```
[the remark this answers is not in the document]
Comment (albanD) [reply aten/src/ATen/native/MaxUnpooling.cpp:275]: The autograd engine guarantees these today. So no need to preserve these.

File: aten/src/ATen/native/cpu/MaxUnpoolKernel.cpp [hunk 3]
```diff
@@ -326,42 +326,9 @@ void max_unpool3d_kernel_impl(
...
-  switch(grad_output.suggest_memory_format()) {
-    case at::MemoryFormat::Contiguous: {
-      AT_DISPATCH_FLOATING_TYPES(grad_output.scalar_type(), "max_unpool2d_backward", [&] {
-        cpu_max_unpool_backward<scalar_t, /*is_3d*/false>(grad_input, grad_output, indices);
-      });
-      break;
-    }
-    case at::MemoryFormat::ChannelsLast: {
-      AT_DISPATCH_FLOATING_TYPES(grad_output.scalar_type(), "max_unpool2d_backward_channels_last", [&] {
-        cpu_max_unpool_backward_channels_last<scalar_t>(grad_input, grad_output, indices);
```
Comment (albanD) [inline aten/src/ATen/native/cpu/MaxUnpoolKernel.cpp:342]: Can this function be removed as well?

File: aten/src/ATen/native/MaxUnpooling.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikitaved) [reply aten/src/ATen/native/MaxUnpooling.cpp:275]: Makes my life easier ;) Great!

File: aten/src/ATen/native/cpu/MaxUnpoolKernel.cpp (hunk 3 above)
Comment (nikitaved) [reply aten/src/ATen/native/cpu/MaxUnpoolKernel.cpp:342]: Good catch. Yes!

**Review by albanD — COMMENTED** [review]
Sounds good.
BC-test needs to be silenced.

Also do you have a quick benchmark to make sure this is not too bad perf-wise?

**Comment by nikitaved** [comment]
A quick benchmark doing something like this:
```python
In [2]: def gen_input(*shape, device='cpu'):
    ...:     x = torch.rand(*shape, dtype=torch.double)
    ...:     pool, indices = torch.nn.functional.max_pool3d(x, (4, 4, 4), return_indices=True)
    ...:     pool.requires_grad_(True)
    ...:     unpool = torch.nn.functional.max_unpool3d(pool, indices, (4, 4, 4))
    ...:     dummy_grad = torch.rand_like(unpool)
    ...:     return [unpool], [dummy_grad]
...
%timeit torch.manual_seed(13); torch.autograd.backward(*gen_input(8, 8, 128, 128, device='cuda'), retain_graph=True); torch.cuda.synchronize()
...
```
Produces:
```
CPU:
shape, master, PR
[8, 16, 16, 16], 893 µs, 798 µs
[8, 32, 32, 32], 4.11 ms, 4.07 ms,
[8, 16, 64, 64], 11.8 ms, 8.46 ms,
[8, 8, 128, 128], 21.6 ms, 17.4 ms,

GPU:
shape, master, PR
[8, 16, 16, 16], 724 µs, 672 µs,
[8, 32, 32, 32], 4.65 ms, 3.87 ms,
[8, 16, 64, 64], 11.4 ms, 6.04 ms,
[8, 8, 128, 128], 18.3 ms, 17.4 ms
```

**Comment by nikitaved** [comment]
@albanD , we have 3 CI failures of which 2 indicate changes in `native_functions.yaml`, and the remaining is not relevant. Unless we have anything else to do, I guess, it is ready now.

**Comment by Lezcano** [comment]
You should add the relevant functions to the skip list for the backwards compat test.

**Comment by nikitaved** [comment]
Oh, there is such a list, did not know.

**Comment by nikitaved** [comment]
@JackCaoG , should the schema change be reflected in XLA?

**Comment by nikitaved** [comment]
@albanD , is it good now?

**Comment by albanD** [comment]
It would be great to have a sister PR on the xla side ready before we land this one. cc @JackCaoG are you the right person to prepare that?

**Comment by JackCaoG** [comment]
Sorry for the delay, it seems like `max_unpool3d_backward`  and `max_unpool2d_backward` is removed and being replaced with some composite backward? These two ops are what pytorch/xla directly lower today, I can try to patch this change and see if backward still works for us.

**Comment by JackCaoG** [comment]
I can confirm that xla build is failing, trying to come up with a pr to fix that.

**Comment by JackCaoG** [comment]
submitted https://github.com/pytorch/xla/pull/3537 for review, test passed locally. PyTorch/XLA CI is having some issue, we are fixing it now.

**Comment by albanD** [comment]
@nikitaved we most likely want to rebase this PR as the CI is now 5 days old while we're confirming the xla fix is good.

**Comment by JackCaoG** [comment]
Still fixing pt/xla CI issue.. will update here when I am done.

**Comment by JackCaoG** [comment]
XLA pr is ready, give me a ping before this one merge. I can merge the xla pr right after.

**Comment by nikitaved** [comment]
@albanD , unless there is something you want to add, I believe we are ready.

**Review by albanD — APPROVED** [review]
SGTM!

**Comment by albanD** [comment]
@pytorchbot merge this please

@JackCaoG can you merge the xla PR when this lands?

**Comment by albanD** [comment]
@nikitaved looks like this needs a rebase :/
To reduce the chances of merge conflict on this file, you can add the entries in alphabetical order ;)

**Comment by JackCaoG** [comment]
I will monitor this pr and merge xla when this one landed.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**